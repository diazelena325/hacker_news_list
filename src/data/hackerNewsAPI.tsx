export const mainUrl = 'https://hacker-news.firebaseio.com/v0/';
export const topStoriesUrl = `${mainUrl}topstories.json`;
export const listingUrl = `${mainUrl}item/`;

//API call to get list of listing Ids of top stories
export async function getListingIds() {
  const result: number[] = await fetch(topStoriesUrl)
    .then(response => response.json())
    .then(data => data);

  return result;
}

//API call to get story details for each listing Id
export async function getEachStory(nsId : number) {
  const result :
    {
      by: string;
      descendants: number;
      id: number;
      kids: number[];
      score: number;
      time: number;
      title: string;
      type: string;
      url: string;
    }
    = await fetch(`${listingUrl + nsId}.json`)
      .then(response => response.json())
      .then((data) => data);

  return result;
}
