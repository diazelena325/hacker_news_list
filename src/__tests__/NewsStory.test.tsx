import { cleanup } from '@testing-library/react';
import { oneStory } from '../fixtures/fixtures';
import { getEachStory } from '../data/hackerNewsAPI';

beforeEach(() => {
    cleanup();
    jest.resetAllMocks();
});

jest.mock('../data/hackerNewsAPI', () => ({
    getEachStory: jest.fn(),
}));
const mockGetEachStory = getEachStory as jest.MockedFunction < typeof getEachStory>;

test('rendering News story component with content', async () => {

    mockGetEachStory.mockResolvedValue(oneStory);
    const getteststory = await getEachStory(1);
    expect(getEachStory).toHaveBeenCalledTimes(1);
    expect(getEachStory).toHaveBeenCalledWith(1)
    expect(getteststory).toMatchSnapshot();
});