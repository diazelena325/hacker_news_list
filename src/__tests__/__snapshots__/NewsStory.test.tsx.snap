// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering News story component with content 1`] = `
Object {
  "by": "Jim Halpert",
  "id": 1,
  "time": 1675362380,
  "title": "Office Technology",
  "url": "https://theoffice.com",
}
`;

